fsspec/spec.py

```diff
@@ -144,7 +144,16 @@ def __init__(self, *args, **storage_options):
         self._intrans = False
         self._transaction = None
         self._invalidated_caches_in_transaction = []
-        self.dircache = DirCache(**storage_options)
+
+        listings_cache_type = storage_options.get("listings_cache_type", "memdircache")
+        if listings_cache_type not in ("memdircache", "filedircache"):
```

Comment:
I would imagine a dircaches dict in which these are registered rather than hardcoded here. External packages might bring their own.

Reply:
Changed this although right now the mapping happens in the `create_dircache` function. Also I created an enum for convenience but this could not be extend by users. What do you think?